src/tyr/route_serializer_osrm.cc

```diff
@@ -1494,93 +1517,94 @@ json::ArrayPtr voice_instructions(const valhalla::DirectionsLeg::Maneuver* prev_
-      voice_instruction_beginning->emplace("ssmlAnnouncement",
-                                           "<speak>" +
-                                               prev_maneuver->verbal_post_transition_instruction() +
-                                               "</speak>");
-      voice_instructions_array->emplace_back(std::move(voice_instruction_beginning));
-    }
-  }
-
-  if (!maneuver.verbal_transition_alert_instruction().empty()) {
-    json::MapPtr voice_instruction_end = json::map({});
+        MIN_DISTANCE_VERBAL_PRE_TRANSITION_INSTRUCTION;
+  }
+  // accordingly we will omit the alert instruction if it is closer than
+  // MIN_DISTANCE_VERBAL_PRE_TRANSITION_INSTRUCTION meters to the maneuver.
+  if (distance_before_verbal_transition_alert_instruction <
+      MIN_DISTANCE_VERBAL_PRE_TRANSITION_INSTRUCTION) {
+    // using -1 here to state that they don't have to be played
+    distance_before_verbal_transition_alert_instruction = -1;
+  }
+
+  if (maneuver_index == 1 && !prev_maneuver->verbal_pre_transition_instruction().empty()) {
+    // For depart maneuver, we always want to hear the verbal_pre_transition_instruction
+    // right at the beginning of the navigation. This is something like:
+    // Drive West on XYZ Street.
+    // This voice_instruction_start is only created once. It is always played, even when
+    // the maneuver would otherwise be too short.
+    addVoiceInstruction(prev_maneuver->verbal_pre_transition_instruction(), double(distance),
+                        voice_instructions);
+  } else if (distance_before_verbal_transition_alert_instruction >= 0.0 &&
+             distance > distance_before_verbal_transition_alert_instruction +
+                            APPROXIMATE_VERBAL_POSTRANSITION_LENGTH &&
+             !prev_maneuver->verbal_post_transition_instruction().empty()) {
+    // In all other cases we want to play the verbal_post_transition_instruction shortly
+    // after the maneuver has started but only if there is sufficient time to play. On
+    // the one hand distance_before_verbal_transition_alert_instruction has to be set,
+    // so there is enough time to play the transition instruction afterwards. On the other hand
+    // there has to be enough time to play the upcoming verbal_pre_transition_instruction and
+    // the verbal_post_transition_instruction itself.
+    // The approximation here is that the verbal_post_transition_instruction takes 100
+    // meters to play + the 10 meters after the maneuver start which is added so that the
+    // instruction is not played directly on the intersection where the maneuver starts.
+    addVoiceInstruction(prev_maneuver->verbal_post_transition_instruction(), double(distance - 10),
+                        voice_instructions);
+  }
+
+  // If there is an alert instruction and we have enough time to play it, we will play it
+  // TODO: We shouldn't add an alert instruction if there was already a pre/post transition alert
+  // which is close by
```

Comment:
Is this still TODO? I'm not quite understanding which cases are still unhandled.

Reply:
There are probably many cases that are not handled. Optimizing is so much about trying out many different szenarios and I think there is still a lot of optimization potential left. This comment was just an idea from my side what still could be improved. Total goal for me with this PR was to make it usable at all, which in my opinion wasn't rally the case before.